### PR TITLE
Prefer error messages to exceptions

### LINKED ISSUE #13: shell_loop should not use throws
Instead of throwing errors, just return them and let rebar handle it upstream. It could show a nice error to the console instead of "something went wrong, use DEBUG=1..." message.

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
[Fix #13].

### CODE REVIEW

File: src/rebar_cmd_prv.erl [hunk 1]
```diff
@@ -35,12 +35,12 @@ do(State) ->
   case lists:keyfind(list_to_atom(CmdName), 1, Config) of
     {_, Command} ->
       rebar_api:debug("Running ~p with command ~p.~n", [[CmdName], [Command]]),
-      case run_shell(Command) of
+      case run_shell({CmdName, Command}) of
         {ok, _} -> {ok, State};
-        Error -> {error, {?MODULE, Error}}
+        {error, Error} -> {error, Error}
```
Comment (paulo-ferraz-oliveira) [inline src/rebar_cmd_prv.erl:40]: This would never happen, since in case of error an exception would be thrown.

File: src/rebar_cmd_prv.erl [hunk 2]
```diff
@@ -35,12 +35,12 @@ do(State) ->
...
       end;
     false ->
-      {error, {?MODULE, no_command}}
+      {error, "Command " ++ CmdName ++ " is unknown. Check your configuration."}
```
Comment (paulo-ferraz-oliveira) [inline src/rebar_cmd_prv.erl:43]: This didn't respect the proposed API (or the `-spec(_).` above it).

**Comment by paulo-ferraz-oliveira** [comment]
Let me know if the messages need to be tweaked.

**Comment by gootik** [comment]
@paulo-ferraz-oliveira This is great thanks for working on it.

**Comment by gootik** [comment]
Going to work on a release in the next few days and get it deployed to hex

**Comment by gootik** [comment]
version 0.2.6 released: https://hex.pm/packages/rebar_cmd/0.2.6

**Pull request merged**